frontend/src/components/AddPlan.tsx

```diff
@@ -0,0 +1,68 @@
+import {useState} from "react";
+import {Plan} from "../types/Plan.ts";
+import axios from "axios";
+type AddPlanProps = {
+    addPlan: (description:string) => void;
+}
+export default function AddPlan({addPlan}:Readonly<AddPlanProps>) {
+
+    const [plans, setPlans] = useState<Plan[]>([]);
+    const [inputVisible, setInputVisible] = useState(false);
+    const [inputValue, setInputValue] = useState('');
+
+    const handleAddClick = () => {
+        setInputVisible(true);
+    };
+
+    const handleSaveClick = () => {
+        if (inputValue.trim() !== '') {
+            const newPlan: Plan= {
+                id: (plans.length + 1).toString(),
```

Comment:
Mit deiner Methoden kann es dazu kommen, dass zwei Pläne die gleiche Id haben. Stell dir vor, du hast drei Pläne in deiner Liste. Der vierte würde dann die id "5" bekommen. Dann löscht du z.B. den zweiten Plan aus der Liste, sodass dein Array nur noch 3 Einträge hat. Wenn du dann wieder einen neuen, vierten Plan hinzufügst, würde dieser auch die id "5" bekommen. 
Benutze hier am besten einer zufällig generierte Id mit UUID()